UOP1_Project/Assets/Scripts/Pool/PoolSO.cs

```diff
@@ -10,17 +10,32 @@ namespace UOP1.Pool
 	/// <typeparam name="T">Specifies the type of elements to pool.</typeparam>
 	public abstract class PoolSO<T> : ScriptableObject, IPool<T>
 	{
-		protected readonly Stack<T> _available = new Stack<T>();
+		protected readonly Stack<T> Available = new Stack<T>();
 		public abstract IFactory<T> Factory { get; set; }
+		protected bool HasBeenPrewarmed { get; set; }
 
 		protected virtual T Create()
 		{
 			return Factory.Create();
 		}
 
+		public virtual void Prewarm(int num)
+		{
+			if (HasBeenPrewarmed)
+			{
+				Debug.LogWarning($"Pool {name} has already been prewarmed.");
+				return;
+			}
```

Comment:
So I made the decision to only allow prewarming of a pool once. However, this doesn't _need_ to be. We could just allow this method to be called and add however many objects to the pool whenever. In that case, it may make sense to call the method `Warm`.